internal/controller/driver_controller.go

```diff
@@ -649,7 +649,7 @@ func (r *driverReconcile) reconcileControllerPluginDeployment() error {
 							})
 						}
 						// Addons Sidecar Container
-						if ptr.Deref(r.driver.Spec.DeployCsiAddons, false) {
+						if !r.isNfsDriver() && ptr.Deref(r.driver.Spec.DeployCsiAddons, false) {
```

Comment:
I think this is unwanted. 
The point of having a user-facing nob is to allow the user the choice. If a user enabled CSI addon on the wrong driver type we should either allow it, or offer some sort of feedback (like rejecting the CR) . 

I can see 2 options here:
1. Remove the change and leave it as user-responsibility to configure correctly
2. Remove the change, but add a step in loadAndValidate to validate to validate the unwanted combinations.

This also has implication on our merge function, we should not merge `DeployCSIAddons = true` if the driver type we are merging into is a not RBD

Reply:
we are doing this in many other places, what is the point to block it here now?

Reply:
To the best of my knowledge, none of the other places are based on user-provided intent (might be wrong here)

Reply:
For example please search for CephFsClientType or Topology

Reply:
I agree, we are failing silently there as well. 
This is what I think we should do: 

1. Lets merge these changes
2. Open a github issue to track the gap we found here

@Madhu-1 WDYT?

Reply:
fine with me.